pkg/dyninst/integration_test.go

```diff
@@ -25,46 +29,958 @@ import (
 
 	"github.com/DataDog/datadog-agent/pkg/dyninst/compiler"
 	"github.com/DataDog/datadog-agent/pkg/dyninst/config"
+	"github.com/DataDog/datadog-agent/pkg/dyninst/decode"
 	"github.com/DataDog/datadog-agent/pkg/dyninst/ir"
 	"github.com/DataDog/datadog-agent/pkg/dyninst/irgen"
 	"github.com/DataDog/datadog-agent/pkg/dyninst/irprinter"
-	object "github.com/DataDog/datadog-agent/pkg/dyninst/object"
+	"github.com/DataDog/datadog-agent/pkg/dyninst/object"
 	"github.com/DataDog/datadog-agent/pkg/dyninst/output"
 	"github.com/DataDog/datadog-agent/pkg/dyninst/testprogs"
-	"github.com/DataDog/datadog-agent/pkg/util/kernel"
 	"github.com/DataDog/datadog-agent/pkg/util/safeelf"
 )
 
-var MinimumKernelVersion = kernel.VersionCode(5, 17, 0)
-
-func skipIfKernelNotSupported(t *testing.T) {
-	curKernelVersion, err := kernel.HostVersion()
-	require.NoError(t, err)
-	if curKernelVersion < MinimumKernelVersion {
-		t.Skipf("Kernel version %v is not supported", curKernelVersion)
-	}
+type testCase struct {
+	name                   string
+	binary                 string
+	probes                 []config.Probe
+	expectedDecodingOutput string
+	expectedToPass         bool
 }
 
 func TestDyninst(t *testing.T) {
-	skipIfKernelNotSupported(t)
-	cfgs := testprogs.GetCommonConfigs(t)
-	for _, cfg := range cfgs {
-		t.Run(cfg.String(), func(t *testing.T) {
+	testCases := []testCase{
```

Comment:
this is going to get out of hand quickly. I think we should replace it with something more snapshot oriented or datadriven sooner rather than later, but perhaps not in this commit or pr.

Reply:
Yea agreed, been mostly for sanity at the moment.